tests/integration_tests/test_spa.py

```diff
@@ -63,8 +101,91 @@ def test_spa_recipe_json_injection():
         soup = BeautifulSoup(f, "lxml")
         assert "https://schema.org" not in str(soup)
 
-        html = inject_recipe_json(str(soup), schema)
+        html = spa.inject_recipe_json(str(soup), schema)
 
     assert "@context" in html
     assert "https://schema.org" in html
     assert recipe_name in html
+
+
+@pytest.mark.parametrize("use_public", [True, False])
+@pytest.mark.asyncio
+async def test_spa_serve_recipe_with_meta(unique_user: TestUser, use_public: bool):
+    recipe = create_recipe(unique_user)
+    user = unique_user.repos.users.get_by_username(unique_user.username)
+    assert user
+
+    response = await spa.serve_recipe_with_meta(
+        user.group_slug, recipe.slug, user=None if use_public else user, session=unique_user.repos.session
+    )
+    assert response.status_code == 200
+    assert "https://schema.org" in response.body.decode()
+
+
+@pytest.mark.parametrize("use_public", [True, False])
+@pytest.mark.asyncio
+async def test_spa_serve_recipe_with_meta_invalid_data(unique_user: TestUser, use_public: bool):
+    recipe = create_recipe(unique_user)
+    user = unique_user.repos.users.get_by_username(unique_user.username)
+    assert user
+
+    response = await spa.serve_recipe_with_meta(
+        random_string(), recipe.slug, user=None if use_public else user, session=unique_user.repos.session
+    )
+    assert response.status_code == 404
+
+    response = await spa.serve_recipe_with_meta(
+        user.group_slug, random_string(), user=None if use_public else user, session=unique_user.repos.session
+    )
+    assert response.status_code == 404
+
+    set_recipe_is_public(unique_user, recipe, is_public=False)
+    response = await spa.serve_recipe_with_meta(
+        user.group_slug, recipe.slug, user=None if use_public else user, session=unique_user.repos.session
+    )
+    if use_public:
+        assert response.status_code == 404
+    else:
+        assert response.status_code == 200
+
+    set_group_is_private(unique_user, is_private=True)
+    set_recipe_is_public(unique_user, recipe, is_public=True)
+    response = await spa.serve_recipe_with_meta(
+        user.group_slug, recipe.slug, user=None if use_public else user, session=unique_user.repos.session
+    )
+    if use_public:
+        assert response.status_code == 404
+    else:
+        assert response.status_code == 200
```

Comment:
Is this implying that a private group's public recipe, is not public? Is that the intention?

Reply:
Yeah that's right, since private group trumps public recipe.

I updated the variable names, "use_public" is confusing